### PR TITLE
Added colorize option for hyper_L grid

### PR DESCRIPTION
**Pull request by ndem0**
I implemented "colorize" option also for the hyper_L grid, in according to the others grids implementations. Now boundary tags is set from 0 to 5, starting from the left boundary and with counterclockwise sense. Just implemented for 2dim. Added small description for the colorize option in the header file.

### CODE REVIEW

File: include/deal.II/grid/grid_generator.h [hunk 1]
```diff
@@ -572,9 +572,12 @@ namespace GridGenerator
    * Initialize the given triangulation with a hyper-L (in 2d or 3d)
    * consisting of exactly <tt>2^dim-1</tt> cells. It produces the hypercube
    * with the interval [<i>left,right</i>] without the hypercube made out of
-   * the interval [<i>(left+right)/2,right</i>] for each coordinate.  All
-   * faces will have boundary indicator 0. This function will create the
-   * classical L-shape in 2d and it will look like the following in 3d:
+   * the interval [<i>(left+right)/2,right</i>] for each coordinate. If the 
+   * @p colorize flag is set, the @p boundary_ids of the surfaces are
+   * assigned, such that the left boundary is 0, and the others are set with
+   * growing number accordingly to the counterclockwise. This function will
```
Comment (Rombur) [inline include/deal.II/grid/grid_generator.h]: Please add a comment that the colorization can only be done in 2D.

File: include/deal.II/grid/grid_generator.h [hunk 2]
```diff
@@ -572,9 +572,12 @@ namespace GridGenerator
    * Initialize the given triangulation with a hyper-L (in 2d or 3d)
    * consisting of exactly <tt>2^dim-1</tt> cells. It produces the hypercube
    * with the interval [<i>left,right</i>] without the hypercube made out of
-   * the interval [<i>(left+right)/2,right</i>] for each coordinate.  All
-   * faces will have boundary indicator 0. This function will create the
-   * classical L-shape in 2d and it will look like the following in 3d:
+   * the interval [<i>(left+right)/2,right</i>] for each coordinate. If the
+   * @p colorize flag is set, the @p boundary_ids of the surfaces are
+   * assigned, such that the left boundary is 0, and the others are set with
+   * growing number accordingly to the counterclockwise. This function will
```
Comment (Rombur) [inline include/deal.II/grid/grid_generator.h]: Add a comment that the colorization only works in 2D.

File: source/grid/grid_generator.cc [hunk 3]
```diff
@@ -350,6 +350,7 @@ namespace GridGenerator
         p1(i) = std::min(p_1(i), p_2(i));
         p2(i) = std::max(p_1(i), p_2(i));
       }
+    //
```
Comment (Rombur) [inline source/grid/grid_generator.cc]: Remove this.

File: source/grid/grid_generator.cc [hunk 4]
```diff
@@ -2108,7 +2110,27 @@ namespace GridGenerator
     tria.create_triangulation (
       std::vector<Point<2> >(&vertices[0], &vertices[8]),
       cells,
-      SubCellData());       // no boundary information
+      SubCellData());
+
+    if (colorize)
+      {
+        auto cell = tria.begin();
```
Comment (Rombur) [inline source/grid/grid_generator.cc]: We don't require C++11 so you cannot use ```auto```.

**Comment by bangerth** [comment]
It would be nice to also have an entry in `doc/news/changes.h` in the "specific changes" section. Would you mind producing a follow-up pull-request for that? Thanks in advance!

**Comment by luca-heltai** [comment]
Nicola, welcome among the contributors! :+1:

**Comment by bangerth** [comment]
Welcome indeed!

**Comment by ndem0** [comment]
Thank you guys, I hope I can continue to help!
Il 25/mar/2016 3:25 AM, "Wolfgang Bangerth" <notifications@github.com> ha
scritto:

**Pull request merged**